src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,883 @@ const ActorInit Object_Kankyo_InitVars = {
+                            this->effects[i].dirPhase.y += 0.05f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.z += 0.015f;
+                            break;
+
+                        case 1:
+                            this->effects[i].dirPhase.x += 0.01f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.y += 0.05f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.z += 0.005f * Rand_ZeroOne();
+                            break;
+
+                        case 2:
+                            this->effects[i].dirPhase.x += 0.01f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.y += 0.4f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.z += 0.004f * Rand_ZeroOne();
+                            break;
+
+                        case 3:
+                            this->effects[i].dirPhase.x += 0.01 * Rand_ZeroOne();
+                            this->effects[i].dirPhase.y += 0.08f * Rand_ZeroOne();
+                            this->effects[i].dirPhase.z += 0.05f * Rand_ZeroOne();
+                            break;
+                    }
+                } else if (this->effects[i].state == 2) {
+                    // scatter when the player moves or after a long enough time
+                    if (playerMoved || this->effects[i].timer > 1280) {
+                        this->effects[i].timer = 0;
+                        this->effects[i].state = 1;
+                        this->effects[i].speed = 1.5f;
+                        this->effects[i].targetSpeed = Rand_ZeroOne() * 1.6f + 0.5f;
+                    }
+
+                    if ((i & sTrailingFairies) == 0) { // leader fairy
+                        Math_SmoothStepToF(&this->effects[i].size, 0.25f, 0.1f, 0.001f, 0.00001f);
+
+                        // move the center of the flight path to player's position
+                        Math_SmoothStepToF(&this->effects[i].base.x, player->actor.world.pos.x, 0.5f, 1.0f, 0.2f);
+                        Math_SmoothStepToF(&this->effects[i].base.y, player->actor.world.pos.y + 50.0f, 0.5f, 1.0f,
+                                           0.2f);
+                        Math_SmoothStepToF(&this->effects[i].base.z, player->actor.world.pos.z, 0.5f, 1.0f, 0.2f);
+
+                        // results unused
+                        Math_SmoothStepToF(&this->effects[i].pos.x,
+                                           Math_SinS(this->effects[i].angle - 0x8000) * this->effects[i].flightRadius,
+                                           0.5f, 2.0f, 0.2f);
+                        Math_SmoothStepToF(&this->effects[i].pos.z,
+                                           Math_CosS(this->effects[i].angle - 0x8000) * this->effects[i].flightRadius,
+                                           0.5f, 2.0f, 0.2f);
+
+                        // the lead fairy flies in a sine wave with y = -cos(dirphase.y + amplitude * t) / amplitude
```

Comment:
it's a rough estimate to give an impression of how `amplitude` affects the way it moves. I agree though that it might be confusing for someone reading this to see `-cos` appear out of nowhere.

Reply:
I would settle with something like
```suggestion
                        // the lead fairy's y position approximately follows a sine wave with `amplitude` as angular frequency and `1 / amplitude` as amplitude
```